src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+        boolean loop = true;
+        System.out.println("Hello! Duke here, how can I help you?");
+        Scanner sc = new Scanner(System.in);
+
+        while (loop) {
+            if (sc.hasNext("bye")) {
+                loop = false;
+                break;
+            }
+
+            if (sc.hasNext("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for (int j = 0; j < list.size(); j++) {
+                    System.out.println(j+1 + ". " + list.get(j).getTask());
+                }
+                sc.nextLine();
+                continue;
+            }
+
+            if (sc.hasNext("mark")) {
+                sc.next();
+                if (sc.hasNextInt()) {
+                    int input = sc.nextInt();
+                    try {
+                        list.get(input - 1).markAsDone();
+                        System.out.println("Good job! Task " + input + " has been completed:");
+                        System.out.println("  " + list.get(input - 1).getTask());
+                    } catch (IndexOutOfBoundsException | NullPointerException e) {
+                        System.out.println("You've given me an invalid task to mark!");
+                    }
+                }
+                sc.nextLine();
+                continue;
+            }
+
+            if (sc.hasNext("unmark")) {
+                sc.next();
+                if (sc.hasNextInt()) {
+                    int input = sc.nextInt();
+                    try {
+                        list.get(input - 1).markAsUndone();
+                        System.out.println("Got it! Task " + input + " has not yet been completed:");
+                        System.out.println("  " + list.get(input - 1).getTask());
+                    } catch (IndexOutOfBoundsException | NullPointerException e) {
+                        System.out.println("You've given me an invalid task to unmark!");
+                    }
+                }
+                sc.nextLine();
+                continue;
+            }
+
+            if (sc.hasNext("delete")) {
+                sc.next();
+                if (sc.hasNextInt()) {
+                    int input = sc.nextInt();
+                    try {
+                        Task removedTask = list.get(input - 1);
+                        list.remove(input - 1);
+                        i--;
+                        System.out.println("Got it! Task " + input + " has been deleted from the list:");
+                        System.out.println("  " + removedTask.getTask());
+                        System.out.println("You have a total of " + i + " tasks in the list.");
+                    } catch (IndexOutOfBoundsException | NullPointerException e) {
+                        System.out.println("You've given me an invalid task to delete!");
+                    }
+                }
+                sc.nextLine();
+                continue;
+            }
+
+            String str = sc.next();
```

Comment:
I think the variable `str` can be representative and descriptive like `userInput`.